lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	GitRepoPath    string
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+func isValidSha(potenitalSHA string) error {
+	sha256Length := 64
+	if len(potenitalSHA) == sha256Length {
+		_, err := hex.DecodeString(potenitalSHA)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	return fmt.Errorf("Length of SHA is %d characters, should be 64",
+		len(potenitalSHA))
+}
+
+// MKRealImageRemovalCheck returns an instance of ImageRemovalCheck.
+func MKRealImageRemovalCheck(
+	gitRepoPath string,
+) (*ImageRemovalCheck, error) {
+	masterSHAStr := os.Getenv("PULL_BASE_SHA")
+	err := isValidSha(masterSHAStr)
+	if err != nil {
+		return nil, fmt.Errorf("The PULL_BASE_SHA environment variable "+
+			"is invalid: %v", err)
+	}
+	pullRequestSHAStr := os.Getenv("PULL_PULL_SHA")
+	err = isValidSha(pullRequestSHAStr)
+	if err != nil {
+		return nil, fmt.Errorf("The PULL_PULL_SHA environment variable "+
+			"is invalid: %v", err)
+	}
+	masterSHA := plumbing.NewHash(masterSHAStr)
+	pullRequestSHA := plumbing.NewHash(pullRequestSHAStr)
+	return &ImageRemovalCheck{
+		gitRepoPath,
+		masterSHA,
+		pullRequestSHA,
+	}, nil
+}
+
+// Run executes ImageRemovalCheck on a set of promotion edges.
+// Returns an error if the pull request removes images from the
+// promoter manifests.
+func (check *ImageRemovalCheck) Run(
+	edges map[PromotionEdge]interface{},
+) error {
+	var err error
+	// Open the Git repo of the current working directory, check
```

Comment:
The first sentence is the most important, and it doesn't explain the *why*, so I think you should remove it.